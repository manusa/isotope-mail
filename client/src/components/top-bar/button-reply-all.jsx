import React from 'react';
import PropTypes from 'prop-types';
import {translate} from 'react-i18next';
import TopBarButton from './top-bar-button';

export const ButtonReplyAll = ({t, outboxEmpty, replyAllMessage}) => (
  outboxEmpty && (
    <span isotip={t('topBar.replyAll')} isotip-position='bottom' isotip-size='small'>
      <TopBarButton onClick={replyAllMessage}>reply_all</TopBarButton>
    </span>)
);

ButtonReplyAll.propTypes = {
  outboxEmpty: PropTypes.bool.isRequired,
  replyAllMessage: PropTypes.func.isRequired
};

export default translate()(ButtonReplyAll);
