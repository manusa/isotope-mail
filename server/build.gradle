buildscript {
	ext {
		springBootVersion = '2.0.4.RELEASE'
	}
	repositories {
		mavenCentral()
	}
	dependencies {
		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
	}
}

apply plugin: 'java'
apply plugin: 'eclipse'
apply plugin: 'org.springframework.boot'
apply plugin: 'io.spring.dependency-management'

group = 'com.marcnuri.isotope'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = 1.8

repositories {
	mavenCentral()
}


dependencies {
	compile('org.springframework.boot:spring-boot-starter')
	compile('org.springframework.boot:spring-boot-starter-web')
	compile('org.springframework.boot:spring-boot-starter-hateoas')
	compile('com.sun.mail:javax.mail:1.6.1')
	testCompile('org.springframework.boot:spring-boot-starter-test')
	testCompile('com.marcnuri:spring-common-test:0.0.2')
}
