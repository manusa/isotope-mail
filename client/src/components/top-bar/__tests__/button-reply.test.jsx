import React from 'react';
import {shallow} from 'enzyme/build/index';
import {ButtonReplyAll} from '../button-reply-all';

describe('ButtonReplyAll component test suite', () => {
  test('Snapshot render, outboxEmpty, should render button', () => {
    // Given
    const props = {t: jest.fn(arg => arg), outboxEmpty: true, replyAllMessage: jest.fn()};
    // When
    const button = shallow(<ButtonReplyAll {...props}/>);
    // Then
    expect(button).toMatchSnapshot();
  });
  test('Snapshot render, not outboxEmpty, should NOT render button', () => {
    // Given
    const props = {t: jest.fn(arg => arg), collapsed: false, replyAllMessage: jest.fn()};
    // When
    const button = shallow(<ButtonReplyAll {...props}/>);
    // Then
    expect(button).toMatchSnapshot();
  });
});
