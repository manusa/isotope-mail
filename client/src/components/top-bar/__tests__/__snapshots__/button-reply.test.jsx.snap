// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonReplyAll component test suite Snapshot render, not outboxEmpty, should NOT render button 1`] = `""`;

exports[`ButtonReplyAll component test suite Snapshot render, outboxEmpty, should render button 1`] = `
<span
  isotip="topBar.replyAll"
  isotip-position="bottom"
  isotip-size="small"
>
  <TopBarButton
    className=""
    onClick={[MockFunction]}
  >
    reply_all
  </TopBarButton>
</span>
`;
